mbedtls/src/cipher/raw/serde.rs

```diff
@@ -209,6 +202,63 @@ impl<'de, Op: Operation> Deserialize<'de> for Cipher<Op, Traditional, CipherData
     }
 }
 
+unsafe fn deserialize_raw_cipher<'de, D>(raw: SavedRawCipher, cipher_context: &mut cipher_context_t)
+    -> Result<(), D::Error>
```

Comment:
Return `Result<(), (&'static str, &'static str)>` instead so you don't have to pass `D` around.